### PR TITLE
[Doc] Add the DCO to CONTRIBUTING.md

### PR DESCRIPTION
**Pull request by russellb**
PRs to vLLM now require a `Signed-off-by` header as an acknowledgement
of the DCO. Add the DCO to the repo and update CONTRIBUTING.md with a
pointer to it.

While in there, add a LICENSE link to the contributing doc as well.

Signed-off-by: Russell Bryant <rbryant@redhat.com>

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -33,6 +35,14 @@ pytest tests/
 
 ## Contribution Guidelines
 
+### DCO and Signed-off-by
+
+When contributing changes to this project, you must agree to the [DCO](DCO).
+Commits must include a `Signed-off-by:` header which certifies agreement with
+the terms of the [DCO](DCO).
+
+Using `-s` with `git commit` will automtaically add this header.
```
Comment (DarkLight1337) [inline CONTRIBUTING.md:44]:
```suggestion
Using `-s` with `git commit` will automatically add this header.
```

Comment (DarkLight1337) [reply CONTRIBUTING.md:44]: Maybe also add a link to https://docs.github.com/en/authentication/managing-commit-signature-verification/signing-commits?

**Review by DarkLight1337 — COMMENTED** [review]
Thanks for adding this - various PR authors have already expressed confusion over the DCO bot, so this should help clear that up!

**Comment by njhill** [comment]
Thanks @russellb this was very much needed!

File: CONTRIBUTING.md (hunk 1 above)
Comment (russellb) [reply CONTRIBUTING.md:44]: Commit signing is different than the `Signed-off-by` bit. Signing with gpg isn't necessary to meet the requirement. (at least in every project I've worked on that uses the DCO)

**Review by mgoin — APPROVED** [review]

File: CONTRIBUTING.md (hunk 1 above)
Comment (DarkLight1337) [reply CONTRIBUTING.md]: Oh, I got confused because I ran into this issue when trying to set up auto sign on my IDE.

**Pull request merged**